### PR TITLE
[CON-7030] maintenance: update k8s dependencies to latest patch version

### PR DESCRIPTION
**Pull request by mo-rieger**
go get k8s.io/api@v0.24.3 k8s.io/apimachinery@v0.24.3 k8s.io/client-go@v0.24.3 k8s.io/cloud-provider@v0.24.3 k8s.io/component-base@v0.24.3
go mod tidy
go mod vendor

CLOSES: CON-7030

### CODE REVIEW

**Comment by timoreimann** [comment]
@mo-rieger looks like our format checker is complaining about two files not being gofmt'ed: https://github.com/digitalocean/digitalocean-cloud-controller-manager/runs/7785308528?check_suite_focus=true#step:5:151

This seems unrelated to your PR and must have slipped through to master. Would you mind pushing a quick commit to fix it?

**Comment by mo-rieger** [comment]
> 

The linting issue occurs only in the CI because set-up go installs the latest minor version of go. I fixed it to use the latest patch version of the go version used in the project.

**Review by timoreimann — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -13,14 +13,14 @@ jobs:
...
-      - name: Install go
-        uses: actions/setup-go@0caeaed6fd66a828038c2da3c0f662a42862658f
-        with:
-          go-version: ^1.17
       - name: Check out code into the Go module directory
         uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
         with:
           ref: master
+      - name: Install go
+        uses: actions/setup-go@v3.1.0
```
[the remark this answers is not in the document]
Comment (mo-rieger) [reply .github/workflows/release.yml]: Got it, changed it to use the commit SHA.

**Review by timoreimann — APPROVED** [review]
Thanks Moritz.

**Review by cpanato — APPROVED** [review]

**Pull request merged**